src/Storage/Storage.Management/StorageAccount/NewAzureStorageAccount.cs

```diff
@@ -863,10 +869,31 @@ public override void ExecuteCmdlet()
                     Name = this.EdgeZone
                 };
             }
-            if (sasExpirationPeriod != null)
+            if (sasExpirationPeriod != null || SasExpirationAction != null)
             {
-                createParameters.SasPolicy = new SasPolicy(sasExpirationPeriod.Value.ToString(@"d\.hh\:mm\:ss"), "Log");
+                if (sasExpirationPeriod == null && SasExpirationAction != null)
+                {
+                    throw new ArgumentException("-SasExpirationAction can only be specified together with -SasExpirationPeriod.", "SasExpirationAction");
+                }
+                // Set the default action to Log to be aligned as before PSH release.
+                if (SasExpirationAction == null)
```

Comment:
This variable has a default value. Is it necessary to have null check here?

Reply:
I believe when you don't input "SasExpirationAction", it will be null:

![image](https://github.com/user-attachments/assets/b1eb7b65-974c-4567-a392-03667fc55f28)

It looks `[PSDefaultValue(Help = "Log", Value = StorageModels.ExpirationAction.Log)]` not work, or it only update help?

Reply:
Yes, it is for doc only. Can keep it there.